### PR TITLE
Feat: list enabled addon from cluster interface and update addon

### PR DESCRIPTION
**Pull request by wangyikewxgm**
### Description of your changes

1. apiserver interface to list enabled addon from cluster 
2. apiserver interface to update addon

Fixes #

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Review by hongchaodeng — APPROVED** [review]

File: pkg/apiserver/rest/usecase/addon.go [hunk 1]
```diff
@@ -402,6 +405,66 @@ func (u *addonUsecaseImpl) DisableAddon(ctx context.Context, name string) error
 	return nil
 }
 
+func (u *addonUsecaseImpl) ListEnabledAddon(ctx context.Context) ([]*apis.AddonStatusResponse, error) {
+	apps := &v1beta1.ApplicationList{}
+	if err := u.kubeClient.List(ctx, apps, client.InNamespace(types.DefaultKubeVelaNS)); err != nil {
```
Comment (barnettZQG) [inline pkg/apiserver/rest/usecase/addon.go:410]: filter by labels?

File: pkg/apiserver/rest/webservice/addon.go [hunk 2]
```diff
@@ -96,6 +103,15 @@ func (s *addonWebService) GetWebService() *restful.WebService {
 		Param(ws.PathParameter("name", "addon name to enable").DataType("string").Required(true)).
 		Writes(apis.AddonStatusResponse{}))
 
+	// update addon
+	ws.Route(ws.POST("/{name}/update").To(s.updateAddon).
```
Comment (barnettZQG) [inline pkg/apiserver/rest/webservice/addon.go:107]:
```suggestion
	ws.Route(ws.PUT("/{name}").To(s.updateAddon).
```

File: pkg/apiserver/rest/webservice/addon.go [hunk 3]
```diff
@@ -96,6 +103,15 @@ func (s *addonWebService) GetWebService() *restful.WebService {
...
+		Doc("update an addon").
```
Comment (barnettZQG) [inline pkg/apiserver/rest/webservice/addon.go:108]: Please define the request body.

File: pkg/apiserver/rest/webservice/addon.go [hunk 4]
```diff
@@ -185,3 +201,69 @@ func (s *addonWebService) statusAddon(req *restful.Request, res *restful.Respons
...
+}
+
+func (s *enabledAddonWebService) GetWebService() *restful.WebService {
+	ws := new(restful.WebService)
+	ws.Path(versionPrefix+"/enabled-addon").
+		Consumes(restful.MIME_XML, restful.MIME_JSON).
+		Produces(restful.MIME_JSON, restful.MIME_XML).
+		Doc("api for addon management")
+
+	tags := []string{"enabled-addon"}
```
Comment (barnettZQG) [inline pkg/apiserver/rest/webservice/addon.go:242]:
```suggestion
	tags := []string{"addon"}
```

**Review by chivalryq — APPROVED** [review]
LGTM

**Review by wonderflow — APPROVED** [review]

**Review by barnettZQG — APPROVED** [review]

**Pull request merged**